app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala

```diff
@@ -0,0 +1,121 @@
+                                    cpd:                           CommonPlayDependencies,
+                                    mcc:                           MessagesControllerComponents,
+                                    errorHandler:                  ErrorHandler,
+                                    reminderFrequencySet:          reminder_frequency_set,
+                                    reminderConfirmation:          reminder_confirmation)(implicit val crypto: Crypto,
+                                                                                          implicit val transformer:                 NINOLogMessageTransformer,
+                                                                                          implicit val reminderFrequencyValidation: ReminderFrequencyValidation,
+                                                                                          val frontendAppConfig:                    FrontendAppConfig,
+                                                                                          val config:                               Configuration,
+                                                                                          val env:                                  Environment,
+                                                                                          ec:                                       ExecutionContext)
+
+  extends BaseController(cpd, mcc, errorHandler) with HelpToSaveAuth with SessionBehaviour with Logging {
+
+  private def backLink: String = routes.AccessAccountController.accessAccount().url
+
+  def getSelectRendersPage(): Action[AnyContent] = authorisedForHtsWithNINO { implicit request ⇒ implicit htsContext ⇒
+
+    Ok(reminderFrequencySet(ReminderForm.giveRemindersDetailsForm(), Some(backLink)))
+
+  }(loginContinueURL = routes.ReminderController.selectRemindersSubmit().url)
+
+  def selectRemindersSubmit(): Action[AnyContent] = authorisedForHtsWithInfo { implicit request ⇒ implicit htsContext ⇒
+
+    ReminderForm.giveRemindersDetailsForm().bindFromRequest().fold(
+      withErrors ⇒ {
+        Ok(reminderFrequencySet(withErrors))
+      },
+
+      success ⇒
+        {
+          //Get the user name
+          val userName = htsContext.userDetails match {
+            case Left(x)  ⇒ ""
+            case Right(x) ⇒ x.forename
+
+          }
+
+          val htsUserModified = for {
+
+            emailRetrieved ← helpToSaveService.getConfirmedEmail
+
+            htsUserModified ← helpToSaveReminderService.updateHtsUser(HtsUser(Nino(htsContext.nino), emailRetrieved.getOrElse("noEmail"), userName, daysToReceive = DateToDaysMapper.d2dMapper.getOrElse(success.reminderFrequency, Seq(1))))
+
+          } yield htsUserModified
+
+          htsUserModified.fold(
+            { e ⇒
+              logger.warn(s"Could not find confirmed email: $e")
+              SeeOther(routes.RegisterController.getServiceUnavailablePage().url)
```

Comment:
No not service unavailable, that is used when the service is unavailable.  This is an internal error, use uk.gov.hmrc.helptosavefrontend.controllers.BaseController#internalServerError

Reply:
Almost all the issues addressed except default value for getEmail(). I will create a technical debt and continue to work on that. Kindly review the code now.